rollup/sync_service/sync_service.go

```diff
@@ -139,6 +141,16 @@ func (s *SyncService) Stop() {
 	}
 }
 
+// ResetStartSyncHeight resets the SyncService to a specific L1 block height
+func (s *SyncService) ResetStartSyncHeight(height uint64) {
+	if s == nil {
+		return
+	}
+
+	s.latestProcessedBlock.Store(height)
```

Comment:
The problem with using atomic is that this update might be overwritten in `fetchMessages`.

Reply:
changed to use cas in [20805d0](https://github.com/scroll-tech/go-ethereum/pull/1044/commits/20805d0ad8700dfe785ea3ffac6d43ae31b8529a).